tasks/agent-macos.yml

```diff
@@ -0,0 +1,175 @@
+---
+- name: 'Load user data into JSON string {"uid": <number>, "gid": <number>}'
+  shell:
+    cmd: "dscacheutil -q user -a name {{ ansible_user }} | awk 'BEGIN { RS=\"\\n\"; ORS=\" \" } /uid:/ { print \"{ \\\"uid\\\": \" $2\",\" } /gid:/ { print \"\\\"gid\\\": \" $2 \" }\"}'"
+    executable: /bin/bash
+  changed_when: false
+  register: macos_user_output
```

Comment:
i've chosen not to parameterize the user/group - the playbook will use the ssh session user (`ansible_user`)

Reply:
While I understand that this way is good for your usecase, it's inconsistent with the rest of the role and might not be good enough for everyone. I think we should really make this configurable.

Reply:
i can't personally imagine the use case, but it is good to have the option.